benchcab/task.py

```diff
@@ -214,10 +230,37 @@ def get_fluxnet_tasks(
             branch_patch=branch["patch"],
             met_forcing_file=site,
             sci_conf_id=sci_conf_id,
-            sci_config=sci_config
+            sci_config=sci_config,
         )
         for branch_id, branch in enumerate(realisations)
         for site in met_sites
         for sci_conf_id, sci_config in enumerate(science_configurations)
     ]
     return tasks
+
+
+def get_fluxnet_comparisons(tasks: list[Task]) -> list[tuple[Task, Task]]:
+    """Returns a list of pairs of fluxnet tasks to run comparisons with.
+
+    Pairs should be matching in science configurations and meteorological
+    forcing, but differ in realisations. When multiple realisations are
+    specified, return all pair wise combinations between all realisations.
+    """
+    return [
+        (task_a, task_b)
+        for task_a in tasks
+        for task_b in tasks
+        if task_a.met_forcing_file == task_b.met_forcing_file
+        and task_a.sci_conf_id == task_b.sci_conf_id
+        and task_a.branch_id < task_b.branch_id
```

Comment:
Should we add a test on `branch_patch` so we don't run the test if only one of the tasks has a patch applied? Since in this case, we are fairly certain the science configurations are different.

If both branches have a patch, we should keep the comparison as the patches could be identical.

Reply:
Yep I agree with those cases where a user does not need to run a regression test. However, I think the best approach would be for the user manually switch off the regression test (we have an issue for this: #77).

It might be counter productive trying to guess when the user wants to switch on/off the regression test. For example, we might want to do a regression test as a sanity check for example.